### DISCUSSION TITLE
FATAL: window and/or OpenGL context creation failed error (Ubuntu)

### DISCUSSION THREAD

**Discussion by Giannis95** [GLVis/glvis#323] [Q&A]
Hi,
 
I installed GLVis and mfem with spack in ubuntu. I am trying to visualise some meshes  with the following command **"glvis -m mfem/data/star.mesh"** I am getting the following error:

**"FATAL: window and/or OpenGL context creation failed"** 

My OpenGl version is 4.6. I have also build it without spack but the same issue persists. Does anyone had the same issue and/or any suggestion what might be the issue ? 

Thank you in advance for any help

**Comment by tzanio** [comment]
Can you run simple OpenGL applications, like `glxgears`?

**Reply by Giannis95** [reply]
Yes, simple applications such  xclock, glxgears work

**Comment by v-dobrev** [comment]
If `glxgears` works, you can try running GLVis with the extra flag `-oldgl`.

**Reply by Giannis95** [reply]
I have tried it, and I still get the same issue that the server is running but when I pass data to it I get that error. I have also tried running the server on a different port, but still I am getting the same error message

**Comment by tzanio** [comment]
Just to check -- does it work in non-server mode, e.g. `./glvis -oldgl -m ../mfem/data/escher.mesh`?

**Reply by Giannis95** [reply]
hello ! yes it works in non-server mode , however the fonts are not visible just on the escher.mesh. Other meshes suchs as beam-tet.mesh are working with the correct fonts in the non-server mode. Thank you for all the help 
![image](https://github.com/user-attachments/assets/0f4df8c8-aef0-4322-bb5c-b18f965eb1b0)

**Reply by tzanio** [reply]
This looks like an OpenGL/SDL/FontConfig issue on your machine... I am not sure what exactly, maybe someone that uses Ubuntu regularly like @najlkin or @v-dobrev can help.

As an alternative, you can use the  [web version of GLVis](https://glvis.org/live/) both to visualize saved data (from a server run with `glvis -saved`) or as  server, see some instructions [here](https://github.com/GLVis/glvis-js?tab=readme-ov-file#serving-to-a-device-on-your-local-network).

**Reply by Giannis95** [reply]
Thank you
